### PR TITLE
[Resolves #1148] Correct path logic

### LINKED ISSUE #1148: Sceptre fails to find template if sceptre_dir is set
### Subject of the issue

When a _relative path_ is passed to `--dir` and `template_path` contains a path relative to `sceptre_dir + "/templates"`, Sceptre builds an  incorrect path to the template.

### Your environment
* version of sceptre - latest master branch
* version of python (python --version) - 3.6.9
* which OS/distro - Ubuntu

### Steps to reproduce

A config like:

```yaml
# redshift-cluster/config/redshift-cluster.yaml
template_path: redshift-cluster.template.yaml.j2
```

and:

```text
tree redshift-cluster/templates/
redshift-cluster/templates/
└── redshift-cluster.template.yaml.j2
```

Run:

```text
sceptre --debug --dir=redshift-cluster validate redshift-cluster.yaml
```

### Expected behaviour

Should work fine and validate the stack.

### Actual behaviour

```text
  File "/usr/local/lib/python3.6/dist-packages/sceptre/template.py", line 90, in body                                                                                                                 
    body = handler.handle()                                                                                                                                                                           
  File "/usr/local/lib/python3.6/dist-packages/sceptre/template_handlers/file.py", line 54, in handle                                                                                                 
    raise e                                                                                                                                                                                           
  File "/usr/local/lib/python3.6/dist-packages/sceptre/template_handlers/file.py", line 43, in handle                                                                                                 
    with open(path) as template_file:                                                                                                                                                                 
FileNotFoundError: [Errno 2] No such file or directory: 'redshift-cluster/templates/redshift-cluster/templates/redshift-iam-roles.template.yaml'                                                      
```

### PR DESCRIPTION
**Pull request by alexharv074**
This resolves a bug introduced in 5c0ab39. 

In the case of `--dir`, the `File` handler class's `handle` method has
received a path relative to the `project_path` due to the previously
incorrectly-named method in the `ConfigReader` class
`_get_absolute_template_path` returning the relative path.

This patch renames the incorrectly-named method
`_get_absolute_template_path` as `_get_relative_template_path` and
corrects the logic in the `File` class's `handle` method.

## PR Checklist

- [X] Wrote a good commit message & description [see guide below].
- [X] Commit message starts with `[Resolve #issue-number]`.
- [ ] Added/Updated unit tests.
- [ ] Added/Updated integration tests (if applicable).
- [X] All unit tests (`make test`) are passing.
- [X] Used the same coding conventions as the rest of the project.
- [X] The new code passes pre-commit validations (`pre-commit run --all-files`).
- [X] The PR relates to _only_ one subject with a clear title.
      and description in grammatically correct, complete sentences.

## Approver/Reviewer Checklist

- [ ] Before merge squash related commits.

## Other Information

[Guide to writing a good commit](http://chris.beams.io/posts/git-commit/)

### CODE REVIEW

**Comment by alexharv074** [comment]
Note also that prior to
introduction of template handlers, a hack had existed to solve the
problem of computing the path when `--dir` is specified, see
https://github.com/Sceptre/sceptre/blob/354e87e8fbf2dafd2692c7c87abc1ecc35ea3f40/sceptre/template.py#L152-L167

File: sceptre/template_handlers/file.py [hunk 1]
```diff
@@ -24,12 +24,8 @@ def schema(self):
         }
 
     def handle(self):
-        project_path = self.stack_group_config.get("project_path")
         input_path = Path(self.arguments["path"])
-        if input_path.is_absolute():
-            path = str(input_path)
-        else:
-            path = str(Path(project_path) / 'templates' / input_path)
```
Comment (jfalkenstein) [inline sceptre/template_handlers/file.py:32]: @alexharv074, you don't want to cut this out. The whole point is that we WANT to support absolute paths... for paths that are absolute; But _relative_ paths should resolve to the sceptre_dir/templates directory.

File: sceptre/template_handlers/file.py [hunk 2]
```diff
@@ -24,12 +24,8 @@ def schema(self):
         }
 
     def handle(self):
-        project_path = self.stack_group_config.get("project_path")
         input_path = Path(self.arguments["path"])
-        if input_path.is_absolute():
```
Comment (jfalkenstein) [inline sceptre/template_handlers/file.py:29]: I think what we might want to check would be something like
```
project_templates = (Path(self.stack_group_config.get("project_path")) / 'templates').absolute()
if input_path.absolute().exists():
    path = str(input_path)
elif (project_templates / input_path).exists():
    path = str(project_templates / input_path)
else:
    # the file doesn't exist... so we should probably blow up?
```

File: sceptre/template_handlers/file.py (hunk 1 above)
Comment (alexharv074) [reply sceptre/template_handlers/file.py:32]: @jfalkenstein My observation was that in both cases, `input_path` will already contain the resolved path by this stage, but let me play some more to see if I can find a counter example.

**Comment by alexharv074** [comment]
## Testing

### Testing legacy `template_path`

|template_path|--dir|cwd|Result 2.6.3|New result|
|---|---|---|---|---|
|redshift-cluster.template.yaml.j2|redshift-cluster|/src|OK|OK|
|redshift-cluster.template.yaml.j2|redshift-cluster|/src/redshift-cluster|`"Check 'redshift-cluster/config' exists."`|`"Check 'redshift-cluster/config' exists."`|
|redshift-cluster.template.yaml.j2|redshift-cluster|/tmp|`"Check 'redshift-cluster/config' exists."`|`"Check 'redshift-cluster/config' exists."`|
|redshift-cluster.template.yaml.j2|NONE|/src|`"Check '/src/config' exists."`|`"Check '/src/config' exists."`|
|redshift-cluster.template.yaml.j2|NONE|/src/redshift-cluster|OK|OK|
|redshift-cluster.template.yaml.j2|NONE|/tmp|`"Check '/tmp/config' exists."`|`"Check '/tmp/config' exists."`|
|redshift-cluster.template.yaml.j2|/src/redshift-cluster|/src|OK|OK|
|redshift-cluster.template.yaml.j2|/src/redshift-cluster|/src/redshift-cluster|OK|OK|
|redshift-cluster.template.yaml.j2|/src/redshift-cluster|/tmp|OK|OK|
|redshift-cluster.template.yaml.j2|/asdfasdf|/src|`"Check '/asdfasdf/config' exists."`|`"Check '/asdfasdf/config' exists."`|
|redshift-cluster.template.yaml.j2|/asdfasdf|/src/redshift-cluster|`"Check '/asdfasdf/config' exists."`|`"Check '/asdfasdf/config' exists."`|
|redshift-cluster.template.yaml.j2|/asdfasdf|/tmp|`"Check '/asdfasdf/config' exists."`|`"Check '/asdfasdf/config' exists."`|
|/src/redshift-cluster/templates/redshift-cluster.template.yaml.j2|redshift-cluster|/src|OK|OK|
|/src/redshift-cluster/templates/redshift-cluster.template.yaml.j2|redshift-cluster|/src/redshift-cluster|`"Check 'redshift-cluster/config' exists."`|`"Check 'redshift-cluster/config' exists."`|
|/src/redshift-cluster/templates/redshift-cluster.template.yaml.j2|redshift-cluster|/tmp|`"Check 'redshift-cluster/config' exists."`|`"Check 'redshift-cluster/config' exists."`|
|/src/redshift-cluster/templates/redshift-cluster.template.yaml.j2|NONE|/src|`"Check '/src/config' exists."`|`"Check '/src/config' exists."`|
|/src/redshift-cluster/templates/redshift-cluster.template.yaml.j2|NONE|/src/redshift-cluster|OK|OK|
|/src/redshift-cluster/templates/redshift-cluster.template.yaml.j2|NONE|/tmp|`"Check '/tmp/config' exists."`|`"Check '/tmp/config' exists."`|
|/src/redshift-cluster/templates/redshift-cluster.template.yaml.j2|/src/redshift-cluster|/src|OK|OK|
|/src/redshift-cluster/templates/redshift-cluster.template.yaml.j2|/src/redshift-cluster|/src/redshift-cluster|OK|OK|
|/src/redshift-cluster/templates/redshift-cluster.template.yaml.j2|/src/redshift-cluster|/tmp|OK|OK|
|/src/redshift-cluster/templates/redshift-cluster.template.yaml.j2|/asdfasdf|/src|`"Check '/asdfasdf/config' exists."`|`"Check '/asdfasdf/config' exists."`|
|/src/redshift-cluster/templates/redshift-cluster.template.yaml.j2|/asdfasdf|/src/redshift-cluster|`"Check '/asdfasdf/config' exists."`|`"Check '/asdfasdf/config' exists."`|
|/src/redshift-cluster/templates/redshift-cluster.template.yaml.j2|/asdfasdf|/tmp|`"Check '/asdfasdf/config' exists."`|`"Check '/asdfasdf/config' exists."`|
|redshift-cluster/templates/redshift-cluster.template.yaml.j2|redshift-cluster|/src|`"redshift-cluster.template.yaml.j2"`|`"(\"No such template file: '%s'\", PosixPath('redshift-cluster/templates/redshift-cluster/templates/redshift-cluster.template.yaml.j2'))"`|
|redshift-cluster/templates/redshift-cluster.template.yaml.j2|redshift-cluster|/src/redshift-cluster|`"Check 'redshift-cluster/config' exists."`|`"Check 'redshift-cluster/config' exists."`|
|redshift-cluster/templates/redshift-cluster.template.yaml.j2|redshift-cluster|/tmp|`"Check 'redshift-cluster/config' exists."`|`"Check 'redshift-cluster/config' exists."`|
|redshift-cluster/templates/redshift-cluster.template.yaml.j2|NONE|/src|`"Check '/src/config' exists."`|`"Check '/src/config' exists."`|
|redshift-cluster/templates/redshift-cluster.template.yaml.j2|NONE|/src/redshift-cluster|`"redshift-cluster.template.yaml.j2"`|`"(\"No such template file: '%s'\", PosixPath('/src/redshift-cluster/templates/redshift-cluster/templates/redshift-cluster.template.yaml.j2'))"`|
|redshift-cluster/templates/redshift-cluster.template.yaml.j2|NONE|/tmp|`"Check '/tmp/config' exists."`|`"Check '/tmp/config' exists."`|
|redshift-cluster/templates/redshift-cluster.template.yaml.j2|/src/redshift-cluster|/src|`"redshift-cluster.template.yaml.j2"`|`"(\"No such template file: '%s'\", PosixPath('/src/redshift-cluster/templates/redshift-cluster/templates/redshift-cluster.template.yaml.j2'))"`|
|redshift-cluster/templates/redshift-cluster.template.yaml.j2|/src/redshift-cluster|/src/redshift-cluster|`"redshift-cluster.template.yaml.j2"`|`"(\"No such template file: '%s'\", PosixPath('/src/redshift-cluster/templates/redshift-cluster/templates/redshift-cluster.template.yaml.j2'))"`|
|redshift-cluster/templates/redshift-cluster.template.yaml.j2|/src/redshift-cluster|/tmp|`"redshift-cluster.template.yaml.j2"`|`"(\"No such template file: '%s'\", PosixPath('/src/redshift-cluster/templates/redshift-cluster/templates/redshift-cluster.template.yaml.j2'))"`|
|redshift-cluster/templates/redshift-cluster.template.yaml.j2|/asdfasdf|/src|`"Check '/asdfasdf/config' exists."`|`"Check '/asdfasdf/config' exists."`|
|redshift-cluster/templates/redshift-cluster.template.yaml.j2|/asdfasdf|/src/redshift-cluster|`"Check '/asdfasdf/config' exists."`|`"Check '/asdfasdf/config' exists."`|
|redshift-cluster/templates/redshift-cluster.template.yaml.j2|/asdfasdf|/tmp|`"Check '/asdfasdf/config' exists."`|`"Check '/asdfasdf/config' exists."`|



### Testing `template.path`

|template_path|--dir|cwd|Result|New result|
|---|---|---|---|---|
|redshift-cluster.template.yaml.j2|redshift-cluster|/src|`No such file or directory: 'redshift-cluster/templates/redshift-cluster/templates/redshift-iam-roles.template.yaml'`|OK|
|redshift-cluster.template.yaml.j2|redshift-cluster|/src/redshift-cluster|`"Check 'redshift-cluster/config' exists."`|`"Check 'redshift-cluster/config' exists."`|
|redshift-cluster.template.yaml.j2|redshift-cluster|/tmp|`"Check 'redshift-cluster/config' exists."`|`"Check 'redshift-cluster/config' exists."`|
|redshift-cluster.template.yaml.j2|NONE|/src|`"Check '/src/config' exists."`|`"Check '/src/config' exists."`|
|redshift-cluster.template.yaml.j2|NONE|/src/redshift-cluster|OK|OK|
|redshift-cluster.template.yaml.j2|NONE|/tmp|`"Check '/tmp/config' exists."`|`"Check '/tmp/config' exists."`|
|redshift-cluster.template.yaml.j2|/src/redshift-cluster|/src|OK|OK|
|redshift-cluster.template.yaml.j2|/src/redshift-cluster|/src/redshift-cluster|OK|OK|
|redshift-cluster.template.yaml.j2|/src/redshift-cluster|/tmp|OK|OK|
|redshift-cluster.template.yaml.j2|/asdfasdf|/src|`"Check '/asdfasdf/config' exists."`|`"Check '/asdfasdf/config' exists."`|
|redshift-cluster.template.yaml.j2|/asdfasdf|/src/redshift-cluster|`"Check '/asdfasdf/config' exists."`|`"Check '/asdfasdf/config' exists."`|
|redshift-cluster.template.yaml.j2|/asdfasdf|/tmp|`"Check '/asdfasdf/config' exists."`|`"Check '/asdfasdf/config' exists."`|
|/src/redshift-cluster/templates/redshift-cluster.template.yaml.j2|redshift-cluster|/src|`No such file or directory: 'redshift-cluster/templates/redshift-cluster/templates/redshift-iam-roles.template.yaml'`|OK|
|/src/redshift-cluster/templates/redshift-cluster.template.yaml.j2|redshift-cluster|/src/redshift-cluster|`"Check 'redshift-cluster/config' exists."`|`"Check 'redshift-cluster/config' exists."`|
|/src/redshift-cluster/templates/redshift-cluster.template.yaml.j2|redshift-cluster|/tmp|`"Check 'redshift-cluster/config' exists."`|`"Check 'redshift-cluster/config' exists."`|
|/src/redshift-cluster/templates/redshift-cluster.template.yaml.j2|NONE|/src|`"Check '/src/config' exists."`|`"Check '/src/config' exists."`|
|/src/redshift-cluster/templates/redshift-cluster.template.yaml.j2|NONE|/src/redshift-cluster|OK|OK|
|/src/redshift-cluster/templates/redshift-cluster.template.yaml.j2|NONE|/tmp|`"Check '/tmp/config' exists."`|`"Check '/tmp/config' exists."`|
|/src/redshift-cluster/templates/redshift-cluster.template.yaml.j2|/src/redshift-cluster|/src|OK|OK|
|/src/redshift-cluster/templates/redshift-cluster.template.yaml.j2|/src/redshift-cluster|/src/redshift-cluster|OK|OK|
|/src/redshift-cluster/templates/redshift-cluster.template.yaml.j2|/src/redshift-cluster|/tmp|OK|OK|
|/src/redshift-cluster/templates/redshift-cluster.template.yaml.j2|/asdfasdf|/src|`"Check '/asdfasdf/config' exists."`|`"Check '/asdfasdf/config' exists."`|
|/src/redshift-cluster/templates/redshift-cluster.template.yaml.j2|/asdfasdf|/src/redshift-cluster|`"Check '/asdfasdf/config' exists."`|`"Check '/asdfasdf/config' exists."`|
|/src/redshift-cluster/templates/redshift-cluster.template.yaml.j2|/asdfasdf|/tmp|`"Check '/asdfasdf/config' exists."`|`"Check '/asdfasdf/config' exists."`|
|redshift-cluster/templates/redshift-cluster.template.yaml.j2|redshift-cluster|/src|`No such file or directory: 'redshift-cluster/templates/redshift-cluster/templates/redshift-iam-roles.template.yaml'`|`"(\"No such template file: '%s'\", PosixPath('redshift-cluster/templates/redshift-cluster/templates/redshift-cluster.template.yaml.j2'))"`|
|redshift-cluster/templates/redshift-cluster.template.yaml.j2|redshift-cluster|/src/redshift-cluster|`"Check 'redshift-cluster/config' exists."`|`"Check 'redshift-cluster/config' exists."`|
|redshift-cluster/templates/redshift-cluster.template.yaml.j2|redshift-cluster|/tmp|`"Check 'redshift-cluster/config' exists."`|`"Check 'redshift-cluster/config' exists."`|
|redshift-cluster/templates/redshift-cluster.template.yaml.j2|NONE|/src|`"Check '/src/config' exists."`|`"Check '/src/config' exists."`|
|redshift-cluster/templates/redshift-cluster.template.yaml.j2|NONE|/src/redshift-cluster|`"(\"No such template file: '%s'\", PosixPath('/src/redshift-cluster/templates/redshift-cluster/templates/redshift-cluster.template.yaml.j2'))"`|`"(\"No such template file: '%s'\", PosixPath('/src/redshift-cluster/templates/redshift-cluster/templates/redshift-cluster.template.yaml.j2'))"`|
|redshift-cluster/templates/redshift-cluster.template.yaml.j2|NONE|/tmp|`"Check '/tmp/config' exists."`|`"Check '/tmp/config' exists."`|
|redshift-cluster/templates/redshift-cluster.template.yaml.j2|/src/redshift-cluster|/src|`"(\"No such template file: '%s'\", PosixPath('/src/redshift-cluster/templates/redshift-cluster/templates/redshift-cluster.template.yaml.j2'))"`|`"(\"No such template file: '%s'\", PosixPath('/src/redshift-cluster/templates/redshift-cluster/templates/redshift-cluster.template.yaml.j2'))"`|
|redshift-cluster/templates/redshift-cluster.template.yaml.j2|/src/redshift-cluster|/src/redshift-cluster|`"(\"No such template file: '%s'\", PosixPath('/src/redshift-cluster/templates/redshift-cluster/templates/redshift-cluster.template.yaml.j2'))"`|`"(\"No such template file: '%s'\", PosixPath('/src/redshift-cluster/templates/redshift-cluster/templates/redshift-cluster.template.yaml.j2'))"`|
|redshift-cluster/templates/redshift-cluster.template.yaml.j2|/src/redshift-cluster|/tmp|`"(\"No such template file: '%s'\", PosixPath('/src/redshift-cluster/templates/redshift-cluster/templates/redshift-cluster.template.yaml.j2'))"`|`"(\"No such template file: '%s'\", PosixPath('/src/redshift-cluster/templates/redshift-cluster/templates/redshift-cluster.template.yaml.j2'))"`|
|redshift-cluster/templates/redshift-cluster.template.yaml.j2|/asdfasdf|/src|`"Check '/asdfasdf/config' exists."`|`"Check '/asdfasdf/config' exists."`|
|redshift-cluster/templates/redshift-cluster.template.yaml.j2|/asdfasdf|/src/redshift-cluster|`"Check '/asdfasdf/config' exists."`|`"Check '/asdfasdf/config' exists."`|
|redshift-cluster/templates/redshift-cluster.template.yaml.j2|/asdfasdf|/tmp|`"Check '/asdfasdf/config' exists."`|`"Check '/asdfasdf/config' exists."`|

File: sceptre/template_handlers/file.py (hunk 1 above)
Comment (alexharv074) [reply sceptre/template_handlers/file.py:32]: Ok, I did more testing. See below. Refactored, so closing this discussion.

File: sceptre/template_handlers/file.py (hunk 2 above)
Comment (alexharv074) [reply sceptre/template_handlers/file.py:29]: @jfalkenstein A few issues with this notably that the non-existence of the file could simply mean the user has forgotten to put the file there. But I think my refactored version renders this discussion moot. Let me know your thoughts.

**Comment by alexharv074** [comment]
I used this script to automate testing all those combinations of various options
https://gist.github.com/alexharv074/8f7f4e59568fcdcdd5e1afe986f78bb3

**Comment by jfalkenstein** [comment]
@alexharv074, I love the thoroughness of your tests! But have you tested this with windows paths? I suspect that sceptreize_path is meant to make it to a "Sceptre-type" path, which always uses forward slashes, not backslashes.

**Comment by jfalkenstein** [comment]
We should always use pathlib or is.path to form paths, never string joining due to windows support

**Comment by jfalkenstein** [comment]
I think the core requirements that should be met here are:
* It works no matter if *nix or windows style paths work
* An absolute filepath is effectively unchanged
* Relative filepaths are resolved to be absolute paths that are children of project_dir/templates path

**Comment by alexharv074** [comment]
@jfalkenstein I made changes per our discussion in Slack. See what you think now.

File: sceptre/template_handlers/file.py [hunk 3]
```diff
@@ -51,3 +50,9 @@ def handle(self):
         except Exception as e:
             helper.print_template_traceback(path)
             raise e
+
+    def _get_relative_template_path(self, template_path):
+        return path.join(
+            self.stack_group_config.get("project_path"), "templates",
+            sceptreise_path(template_path)
```
[the remark this answers is not in the document]
Comment (alexharv074) [reply sceptre/template_handlers/file.py]: Ok refactored to use `normalise_path` as discussed.

**Review by jfalkenstein — APPROVED** [review]

**Review by zaro0508 — APPROVED** [review]
thanks for fixing @alexharv074.  `template_path` is [deprecated](https://sceptre.cloudreach.com/dev/docs/template_handlers.html), we really just need to remove it in the next major release of sceptre.

**Pull request merged**